api/scripts/neutralize-answers-after-invalid-challenges.js

```diff
@@ -0,0 +1,99 @@
+'use strict';
+require('dotenv').config({ path: `${__dirname}/../.env` });
+const _ = require('lodash');
+const bluebird = require('bluebird');
+
+const { knex } = require('../db/knex-database-connection');
+const logger = require('../lib/infrastructure/logger');
+
+const challengeRepository = require('../lib/infrastructure/repositories/challenge-repository');
+const certificationAssessmentRepository = require('../lib/infrastructure/repositories/certification-assessment-repository');
+const usecases = require('../lib/domain/usecases/index');
+const events = require('../lib/domain/events');
+
+const extractLengthyChallenges = async () => {
+  const challenges = await challengeRepository.findValidated();
+  const lengthyChallenges = _.remove(challenges, function (challenge) {
+    return challenge.validator.solution.value.length >= 500;
+  });
+
+  logger.info(`${lengthyChallenges.length} challenges found`);
+  logger.debug(lengthyChallenges);
+
+  if (lengthyChallenges.length === 0) {
+    throw new Error('No challenges found, ending process');
+  }
+
+  return lengthyChallenges;
+};
+
+const extractAnswers = async (lengthyChallenges, exposure) => {
+  const answers = await knex
+    .from('sessions')
+    .select('certification-courses.id AS certificationCourseId', 'answers.challengeId')
+    .innerJoin('certification-courses', 'certification-courses.sessionId', 'sessions.id')
+    .innerJoin('assessments', 'assessments.certificationCourseId', 'certification-courses.id')
+    .innerJoin('answers', 'answers.assessmentId', 'assessments.id')
+    .where('sessions.date', '=', exposure.date)
+    .where('answers.result', '<>', 'ok')
+    .whereBetween('answers.createdAt', exposure.window)
+    .whereIn('answers.challengeId', lengthyChallenges);
+
+  logger.info(`${answers.length} answers found`);
+  logger.debug(answers);
```

Comment:
A deplacer apres la ligne 47 (le bloc `if (answers.length === 0)`)
```suggestion
  logger.debug({answers});
```

Reply:
Bien vu, corrigé !